gorm/transaction.go

```diff
@@ -45,10 +56,41 @@ func FromContext(ctx context.Context) (txn *Transaction, ok bool) {
 type Transaction struct {
 	mu              sync.Mutex
 	parent          *gorm.DB
+	parentRO        *gorm.DB
 	current         *gorm.DB
+	txOpts          *sql.TxOptions
+	currentDB       dbType
 	afterCommitHook []func(context.Context)
 }
 
+type databaseOptions struct {
+	readOnlyReplica bool
```

Comment:
Consider using `dbType` here (so you can have notset / readonly / readwrite as a trinary instead of boolean value).